lib/train/options.rb

```diff
@@ -59,6 +59,9 @@ def merge_options(base, opts)
           default = hm[:default]
           if default.is_a? Proc
             res[field] = default.call(res)
+          elsif hm.key?(:coerce)
+            field_value = hm[:coerce].call(res[:host])
```

Comment:
I don't think it makes sense in general to pass the hostname to the Proc. Instead, just pass the options hash (`res`)  to the Proc, and let the Proc internally access the field that it needs (if any - here both will need host).

Reply:
Sure i'll make those changes. Thanks @clintoncwolfe !!